src/main/java/by/exadel/internship/service/impl/TimeForCallUserServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+package by.exadel.internship.service.impl;
+
+
+import by.exadel.internship.dto.timeForCall.TimeForCallWithUserDTO;
+import by.exadel.internship.dto.timeForCall.TimeForCallWithUserIdDTO;
+import by.exadel.internship.dto.timeForCall.TimeForCallUserDTO;
+import by.exadel.internship.dto.UserDTO;
+import by.exadel.internship.dto.enums.InterviewTime;
+import by.exadel.internship.mapper.TimeForCallUserMapper;
+import by.exadel.internship.repository.TimeForCallUserRepository;
+import by.exadel.internship.service.TimeForCallUserServise;
+import by.exadel.internship.service.UserService;
+import by.exadel.internship.util.MDCLog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+;
+
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TimeForCallUserServiceImpl implements TimeForCallUserServise {
+
+    private final TimeForCallUserRepository timeForCallUserRepository;
+    private final UserService userService;
+    private final TimeForCallUserMapper mapper;
+
+    private static final int DEFAULT_START_MINUTES_IF_BETWEEN_ZERO_THIRTY = 30;
+    private static final int DEFAULT_START_MINUTES_IF_BETWEEN_THIRTY_ZERO = 0;
+    private static int INTERVIEW_TIME = 30;
+    private static final String SIMPLE_CLASS_NAME = TimeForCallUserServise.class.getSimpleName();
+
+
+    private List<TimeForCallWithUserDTO> resultUSerTimeList;
```

Comment:
Bean `TimeForCallUserServiceImpl` is singleton. It means that Spring IoC container holds only 1 instance of this class. As you're implementing multiuser application all requests will access to the same field `resultUSerTimeList`, but they need their own isolated property. So move it inside methods where it's used to make it thread-safe.

Reply:
done